client/src/main/scala/org/apache/celeborn/client/LifecycleManager.scala

```diff
@@ -692,6 +834,13 @@ class LifecycleManager(appId: String, val conf: CelebornConf) extends RpcEndpoin
     // check if there exists request for the partition, if do just register
     val requests = changePartitionRequests.computeIfAbsent(shuffleId, rpcContextRegisterFunc)
     inBatchPartitions.computeIfAbsent(shuffleId, inBatchShuffleIdRegisterFunc)
+
+    // handle hard split
+    if (batchHandleCommitPartitionEnabled && cause.isDefined && cause.get == StatusCode.HARD_SPLIT) {
+      committedPartitionInfo.get(shuffleId).commitPartitionRequests
```

Comment:
I think we should put all code referring ShuffleCommittedInfo inside ShuffleCommittedInfo.synchronize{}

Reply:
And we can convert ConcurrentHashMap/KeySetView to HashMap/HashSet then.